frontend/include/chpl/resolution/scope-types.h

```diff
@@ -60,6 +61,55 @@ class IdAndFlags {
   /** A bit-set of the flags defined in the above enum */
   using Flags = uint16_t;
 
+  /** A set of bit-sets / flag combinations. Logically, while Flags
+      represents a conjunction of all of its bitfields (PUBLIC | METHOD
+      means PUBLIC /\ METHOD), this FlagSet represents a disjunction of
+      each flag combination. That is to say, if FlagSet contains:
+
+          PRIVATE, PUBLIC | METHOD
+
+      This represents the following condition on variables:
+
+          PRIVATE \/ (PUBLIC /\ METHOD)
+
+      Inserting into the FlagSet automatically tries to perform basic
+      simplificaton to avoid growing the size. */
+  class FlagSet {
+   private:
+     llvm::SmallVector<Flags, 4> flagVec;
+
+   public:
+    FlagSet() = default;
+    FlagSet(const FlagSet& other) = default;
+    FlagSet(FlagSet&& other) = default;
+    FlagSet& operator=(const FlagSet& other) = default;
+    FlagSet& operator=(FlagSet&& other) = default;
+
+    /** Create a FlagSet consisting of only one combination of Flags. */
```

Comment:
Let's adjust this comment to indicate that Flags is a bitset representing a conjuction of flags.